internal/bridge/crunchybridgecluster/crunchybridgecluster_controller.go

```diff
@@ -265,12 +265,13 @@ func (r *CrunchyBridgeClusterReconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{}, nil
 	}
 
-	// Remove cluster invalid status if found
-	if invalid != nil &&
-		invalid.Status == metav1.ConditionFalse &&
-		invalid.Reason == "ClusterInvalid" {
-		meta.RemoveStatusCondition(&crunchybridgecluster.Status.Conditions,
-			v1beta1.ConditionCreating)
+	invalidUpgrade := meta.FindStatusCondition(crunchybridgecluster.Status.Conditions,
```

Comment:
:pick: A comment above this block of code explaining what it is doing would be nice.